### PR TITLE
Add tests that make a hart unavailable while running

### PR DESCRIPTION
**Pull request by timsifive**
Use the `cease` instruction for this. It's unfortunately non-standard, but it does exist in Rocket so it's not crazy to use it here.

### CODE REVIEW

File: debug/programs/entry.S [hunk 1]
```diff
@@ -204,6 +204,18 @@ trap_entry:
...
+  .align 2
+precease:
+    // Loop a while so that OpenOCD might have confirmed the resume before the
+    // hart becomes unavailable.
+    li t1, 100000
+1:
+    addi t1, t1, -1
+    bnez t1, 1b
+cease:
+  .word 0x30500073  // cease
```
Comment (SiFiveGregS) [inline debug/programs/entry.S:216]: So I guess Spike supports the CEASE instruction, that's good

**Review by SiFiveGregS — APPROVED** [review]

File: debug/programs/entry.S (hunk 1 above)
Comment (aswaterman) [reply debug/programs/entry.S:216]: It doesn't; it'll raise an illegal-instruction exception.

Comment (timsifive) [reply debug/programs/entry.S:216]: Right. I've been running this against an FPGA image that does support the cease instruction.

Comment (SiFiveGregS) [reply debug/programs/entry.S:216]: ah, yeah, got it, thanks!

**Comment by timsifive** [comment]
Looks like I already used this branch for a PR that's no longer necessary. (Mostly merged in #434?)

**Pull request closed without merging**